### PR TITLE
Features/search log

### PR DESCRIPTION
**Pull request by nadavbh-moveo**

### CODE REVIEW

File: src/environment/environment.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export const POKEMON_API_URL = 'https://pokeapi.co/api/v2/pokemon/'
```
Comment (kennethmoveo) [inline src/environment/environment.ts:1]: There is another way to define environment variables that also can have variations depending on the environment (values for prod will be different than values from local). Check it here: https://www.digitalocean.com/community/tutorials/angular-environment-variables

**Review by kennethmoveo — CHANGES_REQUESTED** [review]

**Review by sivanbasa93 — COMMENTED** [review]
its better to use one unit of measure in scss files. (choose rem or px and use it consistently)

File: src/app/components/login/login.component.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+export class LoginComponent implements OnInit, OnDestroy{
+
+  isLoggedIn: boolean = false;
+  isShowError: boolean = false
+  errorMsg: string = AUTH_ERROR_MSG
+  loginSub!: Subscription
+
+  constructor(private authService: AuthService, private rpService: RouteParamsService, private router:Router) {}
+  ngOnInit(): void{
+    this.rpService.pokemonTrackId.next(999)
```
[the remark this answers is not in the document]
Comment (nadavbh-moveo) [reply src/app/components/login/login.component.ts]: @kennethmoveo
Done

File: src/app/components/pokemons/pokemon-search-log/pokemon-search-log.component.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  ngOnInit(): void {
+
+    this.loginSub = this.authService.loginStream.subscribe(isLogged => {
+      if (!isLogged) this.router.navigate(['login'])
+    })
+    this.authService.isAlreadyLogged()
+
+    const recentPokemonIds = this.pokemonService.loadPokemonSearchLog()
+    this.routeParamsService.pokemonTrackId.next(999)
```
[the remark this answers is not in the document]
Comment (nadavbh-moveo) [reply src/app/components/pokemons/pokemon-search-log/pokemon-search-log.component.ts]: @kennethmoveo 
Done

File: src/app/services/pokemon.service.ts [hunk 4]
```diff
@@ -3,20 +3,24 @@ import { Injectable } from '@angular/core';
...
   pokemonsChanged: Subject<IPokemon[]> = new Subject();
-  constructor(private http: HttpClient) {}
+  pokemonsSearch: Subject<IPokemon[] | null> = new Subject();
+  recentSearchedPokemons: number[] = this.loadPokemonSearchLog() || [];
+  constructor(private http: HttpClient, private storageService: StorageService) {}
 
   fetchPokemons() {
     this.http
-      .get<IApiObject>('https://pokeapi.co/api/v2/pokemon/?offset=0&limit=100/')
+      .get<IApiObject>(POKEMON_API_URL + '?offset=0&limit=100')
```
[the remark this answers is not in the document]
Comment (nadavbh-moveo) [reply src/app/services/pokemon.service.ts]: @kennethmoveo 
Done

File: src/app/services/pokemon.service.ts [hunk 5]
```diff
@@ -30,43 +34,64 @@ export class PokemonService {
...
+
           this.pokemons = pokemonArray;
           this.pokemonsChanged.next(pokemonArray);
         });
       });
   }
 
   getPokemon(id: number): Observable<IPokemon> {
-    return this.http.get<IPokemon>(`https://pokeapi.co/api/v2/pokemon/${id}`);
+    return this.http.get<IPokemon>(POKEMON_API_URL + id);
```
[the remark this answers is not in the document]
Comment (nadavbh-moveo) [reply src/app/services/pokemon.service.ts]: @kennethmoveo 
Done

File: src/environment/environment.ts (hunk 1 above)
Comment (nadavbh-moveo) [reply src/environment/environment.ts]: @kennethmoveo 
Done

File: src/app/services/pokemon.service.ts [hunk 6]
```diff
@@ -30,43 +34,64 @@ export class PokemonService {
...
       });
     });
     this.pokemonsChanged.next(filteredPokemons);
   }
+
+  savePokemonToSearchLog(pokemonId: number) {
+    if (this.recentSearchedPokemons.includes(pokemonId)) return
+    else if (this.recentSearchedPokemons.length >= 5) this.recentSearchedPokemons.shift();
+    this.recentSearchedPokemons.push(pokemonId);
+    console.log(this.recentSearchedPokemons)
```
[the remark this answers is not in the document]
Comment (nadavbh-moveo) [reply src/app/services/pokemon.service.ts]: @sivanbasa93  
Done (switched rem to px as well)

**Review by kennethmoveo — APPROVED** [review]
Everything looks right, just be sure to not push the environments variables file to the repo. Right now we don't have sensitive information but on other projects, you will have sensitive information in this file so it's not recommended to upload to GitHub.

**Pull request merged**